articles/article-093-rices-theorem.md

```diff
@@ -0,0 +1,40 @@
+<!-- BEGIN TITLE -->
+# Why There Is No Perfect Anti-Virus
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+In computer science, simple "Yes/No" questions are called **decision
+problems**.  A decision problem is called **decidable** if we can
+write a program that can always find an answer for such a problem.
+
+An example for such a decision problem is "does this program contain a
+computer virus".  If this problem was decidable, there could be
+another program that could always correctly answer this question with
+"Yes" or "No".  A mathematician called Henry Gordon Rice showed that
+such decision problems about the behavior of another program are
+always undecidable, unless the answer is **always** the same.
+
+As most viruses are hidden inside other software and do not always
+expose their real nature, this decision problem can't be solved
```

Comment:
As most viruses are hidden inside other software and don't always expose their true intention, they are difficult to decide using Rice's theorem.

Reply:
I agree with most of your proposed changes, except for the last part: you don't decide a virus, and you don't do it using Rice's theorem.